packages/react-components/src/components/TroubleShootingGuideErrorBar.tsx

```diff
@@ -0,0 +1,150 @@
+  messageBarIconProps,
+  messageBarType
+} from './utils';
+
+/**
+ * Strings for {@link _TroubleShootingGuideErrorBar}.
+ *
+ * @internal
+ */
+export interface _TroubleShootingGuideErrorBarStrings {
+  linkText?: string;
+  buttonText?: string;
+}
+
+/**
+ * Props for {@link _TroubleShootingGuideErrorBar}.
+ *
+ * @internal
+ */
+export interface _TroubleShootingGuideErrorBarProps extends ErrorBarProps {
+  /**
+   * permissions state for camera/microphone
+   */
+  permissionsState: {
+    camera: PermissionState;
+    microphone: PermissionState;
+  };
+  /**
+   * Callback you may provide to supply users with further steps to troubleshoot why they have been
+   * unable to grant your site the required permissions for the call.
+   *
+   * @example
+   * ```ts
+   * onPermissionsTroubleshootingClick: () =>
+   *  window.open('https://contoso.com/permissions-troubleshooting', '_blank');
+   * ```
+   *
+   * @remarks
+   * if this is not supplied, the composite will not show a 'further troubleshooting' link.
+   */
+  onPermissionsTroubleshootingClick?: (permissionsState: {
+    camera: PermissionState;
+    microphone: PermissionState;
+  }) => void;
+  /**
+   * Optional callback to supply users with further troubleshooting steps for network issues
+   * experienced when connecting to a call.
+   *
+   * @example
+   * ```ts
+   * onNetworkingTroubleShootingClick?: () =>
+   *  window.open('https://contoso.com/network-troubleshooting', '_blank');
+   * ```
+   *
+   * @remarks
+   * if this is not supplied, the composite will not show a 'network troubleshooting' link.
+   */
+  onNetworkingTroubleShootingClick?: () => void;
+  /**
+   * strings related to trouble shooting guidance link and dismiss button
+   */
+  troubleShootingGuideStrings: _TroubleShootingGuideErrorBarStrings;
+}
+
+/**
+ * @internal
+ * A component to show device Permission/network connection related errors, contains link that leads to further trouble shooting guide
+ */
+export const _TroubleShootingGuideErrorBar = (props: _TroubleShootingGuideErrorBarProps): JSX.Element => {
+  const theme = useTheme();
+  // error bar strings
+  const localeStrings = useLocale().strings.errorBar;
+  const strings = props.strings ?? localeStrings;
+
+  const [dismissedErrors, setDismissedErrors] = useState<DismissedError[]>([]);
+
+  const {
+    onPermissionsTroubleshootingClick,
+    onNetworkingTroubleShootingClick,
+    permissionsState,
+    troubleShootingGuideStrings
+  } = props;
+
+  // dropDismissalsForInactiveErrors only returns a new object if `dismissedErrors` actually changes.
+  // Without this behaviour, this `useEffect` block would cause a render loop.
+  useEffect(
+    () => setDismissedErrors(dropDismissalsForInactiveErrors(props.activeErrorMessages, dismissedErrors)),
+    [props.activeErrorMessages, dismissedErrors]
+  );
+
+  const toShow = errorsToShow(props.activeErrorMessages, dismissedErrors);
+
+  return (
+    <Stack data-ui-id="error-bar-stack">
```

Comment:
Can this use the existing errorBar? And do something like:
```tsx
<ErrorBar onTroubleShootingClick={() => etc...} linkText={troubleShootingGuideStrings.linkText} ...otherProps />
```

These new props could be added as a beta property, or have an <ErrorBarInner /> component such that it has these extra props and errorbar is simply a wrapper around it:
```tsx
const ErrorBar = (props: ErrorBarProps) => <ErrorBarInner {...props} />;
const _TroubleShootingGuideErrorBar = (props) => <ErrorBarInner {...props} onTroubleShootingClick={() => etc...} linkText={troubleShootingGuideStrings.linkText} />;
```

Reply:
We can't use the existing one because we need to replace the X button with new OK button

Reply:
Ahh gotcha

Reply:
Is it not possible to have some kind of condition that would show the x or the ok button?
Looking at FluentUI, it looks like there is a prop for "actions" that can be used to present the okay button:
https://developer.microsoft.com/en-us/fluentui#/controls/web/messagebar
![image](https://user-images.githubusercontent.com/97130533/190470951-506323a0-4813-4d88-80ae-0046e8cb38aa.png)

Reply:
I think we are tyring to avoid changing the error bar API

Reply:
To avoid changing the error bar API you would introduce a `<ErrorBarInner />` that ErrorBar, NetworkErrorBar etc. would all use